doc/reference/networking/can_api.rst

```diff
@@ -0,0 +1,282 @@
+the same voltage level. This state is also the idle state.
+To write a dominant bit to the bus, open-drain transistors tie CAN H to Vdd
+and CAN L to ground.
+The first and last node use a 120-ohm resistor between CAN H and CAN L to
+terminate the bus. The dominant state always overrides the recessive state.
+This structure is called a wired-AND.
+
+.. warning::
+
+   CAN controllers can only initialize when the bus is in the idle (recessive)
+   state for a specific amount of time. Therefore you have to make sure that
+   CAN RX is high, at least for a short time. This is also necessary for
+   loopback mode.
+
+.. image:: can_transceiver.svg
+   :width: 70%
+   :align: center
+   :alt: CAN Transceiver
+
+CAN uses so-called identifiers to identify the frame instead of addresses to
+identify a node.
+This identifier can either have 11-bit width (Standard or Basic Frame) or
+29-bit in case of an Extended Frame. The Zephyr CAN API supports both Standard
+and Extended identifiers concurrently. A CAN frame starts with a dominant
+Start Of Frame bit. After that, the identifiers follow. This phase is called the
+arbitration phase. During the arbitration phase, write collisions are allowed.
+They resolve by the fact that dominant bits override recessive bits.
+Nodes monitor the bus and notice their transmission being overridden
+abort their transmission. It results in the fact that lower number identifiers
+have priority over higher number identifiers.
+
+Filters are used to whitelist identifiers that are of interest for the specific
+node. An identifier that doesn't match any filter is ignored.
+Filters can either match exactly or a specified part of the identifier.
+This method is called masking.
+As an example, a mask with 11 bits set for standard or 29 bits set for extended
+identifiers must match perfectly.
+Bits that are set to zero in the mask don't care for the matching.
+The filters are implemented in hardware for the most CAN controllers.
+The hardware therefor limits the number of filters.
+Additionally, the number of filters is limited in Kconfig, to save memory.
+
+
+`CAN Wikipedia article <https://en.wikipedia.org/wiki/CAN_bus>`_.
+
+Zephyr supports following CAN features:
+
+* Standard and Extended Identifers
+* Filters with Masking
+* Loopback and Silent mode
+* Remote Request
+
+Sending
+*******
+
+The following code snippets show how to send data.
+
+.. code-block:: C
+
+  struct zcan_frame frame = {
+          .id_type = CAN_STANDARD_IDENTIFIER,
+          .rtr = CAN_DATAFRAME,
+          .ext_id = 0x123,
+          .dlc = 8
+  };
+  struct device *can_dev;
+  int ret;
+
+  frame.data = {1,2,3,4,5,6,7,8};
+
+  can_dev = device_get_binding("CAN_0");
+
+  ret = can_send(can_dev, &frame, K_MSEC(100), NULL, NULL);
+  if (ret != CAN_TX_OK) {
+          LOG_ERR("Sending failed [%d]", ret);
+  }
+
+This basic sample sends a CAN frame with standard identifier 0x123 and eight
```

Comment:
I'd recommend putting the description before the sample code (instead of after).

Reply:
Moved all descriptions before the samples.